staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -106,7 +106,18 @@ import (
 // and index updates happen before such a prescribed notification.
 // For a given SharedInformer and client, the notifications are
 // delivered sequentially.  For a given SharedInformer, client, and
-// object ID, the notifications are delivered in order.
+// object ID, the notifications are delivered in order.  Because
+// `ObjectMeta.UID` has no role in identifying objects, if an object
+// O1 with ID (e.g. namespace and name) X and `ObjectMeta.UID` U1 in
+// the SharedInformer's local cache is deleted and shortly after
+// another relevant object O2 with ID X and `ObjectMeta.UID` U2 is
+// created a client has no guarantee of seeing O1's deletion and
+// O2's creation as two separate notifications, it might as well
```

Comment:
I suggest replacing "might as well" with "could instead" or "might instead", because in common usage "it might as well X" (in a context where Y has been discussed) means that "it" could _choose_ to do X and get the same result as if "it" had chosen to do Y but with less effort.  But that is not the situation here, "it" is not choosing whether to do X or Y.